Tests/UnitTests/Networking/Backend/BackendPostReceiptDataTests.swift

```diff
@@ -687,7 +691,45 @@ class BackendPostReceiptDataTests: BaseBackendTests {
 
 }
 
-private extension BackendPostReceiptDataTests {
+// swiftlint:disable:next type_name
+class BackendPostReceiptCustomEntitlementsTests: BaseBackendPostReceiptDataTests {
+
+    override func createClient() -> MockHTTPClient {
+        super.createClient(#file)
+    }
+
+    override var dangerousSettings: DangerousSettings {
+        return .init(autoSyncPurchases: true, customEntitlementComputation: true)
+    }
+
+    func testDoesNotPostConsentStatus() throws {
+        let path: HTTPRequest.Path = .postReceiptData
+
+        self.httpClient.mock(
+            requestPath: path,
+            response: .init(statusCode: .success, response: Self.validCustomerResponse)
+        )
+
+        waitUntil { completed in
+            self.backend.post(receiptData: Self.receiptData,
+                              appUserID: Self.userID,
+                              isRestore: false,
+                              productData: nil,
+                              presentedOfferingIdentifier: nil,
+                              observerMode: false,
+                              initiationSource: .queue,
+                              subscriberAttributes: nil,
+                              completion: { _ in
+                completed()
+            })
+        }
+
+        expect(self.httpClient.calls).to(haveCount(1))
```

Comment:
I might be missing something, but I don't see anything here specifically checking that attConsentStatus isn't sent. Are we relying on snapshot testing for that part?

Reply:
Yeah. Do you think that's too implicit and we should have something more clear? It would require introspecting the `[String: Any]` of the request post. We didn't have any explicit tests for the opposite case either.

Reply:
I was a bit confused about it when I read the test name tbh, since the test doesn't actually check for it. 
I think it's probably nice to have it since it might be clearer if you mess up the code, when you see exactly what test failed and how, but it's not the end of the world

Reply:
>>> rror: -[UnitTests.BackendPostReceiptCustomEntitlementsTests testDoesNotPostConsentStatus] : failed - Snapshot does not match reference.

@−
"Tests/UnitTests/Networking/Backend/__Snapshots__/BackendPostReceiptDataTests/iOS16-testDoesNotPostConsentStatus.1.json"
@+
"BackendPostReceiptDataTests/iOS16-testDoesNotPostConsentStatus.1.json"

@@ −4,8 +4,14 @@
   },
   "request" : {
     "body" : {
       "app_user_id" : "user",
+      "attributes" : {
+        "$attConsentStatus" : {
+          "updated_at_ms" : 1678307200000,
+          "value" : "authorized"
+        }
+      },
       "fetch_token" : "YW4gYXdlc29tZSByZWNlaXB0",
       "initiation_source" : "queue",
       "is_restore" : false,
       "observer_mode" : false